pkg/pillar/cmd/baseosmgr/worker.go

```diff
@@ -18,99 +18,18 @@ type installWorkDescription struct {
 	target    string
 }
 
-// installWorkResult result of sending to a partition
-type installWorkResult struct {
-	worker.WorkResult // Error etc
-}
-
-var pendingInstallMap = make(map[string]bool)
-var installWorkResultMap = make(map[string]installWorkResult)
-
-func lookupPendingInstall(key string) bool {
-	res, ok := pendingInstallMap[key]
-	return ok && res
-}
-
-func addPendingInstall(key string) {
-	pendingInstallMap[key] = true
-}
-
-func deletePendingInstall(key string) {
-	delete(pendingInstallMap, key)
-}
-
 // AddWorkInstall checks if the Key is in the map of pending work
 // and if not kicks of a worker and adds it
 func AddWorkInstall(ctx *baseOsMgrContext, key, ref, target string) {
-	log.Infof("AddWorkInstall(%s)", key)
-	if lookupPendingInstall(key) {
```

Comment:
There might be cases where baseosmgr/volumemgr does an Add without tracking whether it has already done an Add. For volumemgr this might not be the case as the state transitions to CREATING_VOLUME but I don't recall such a state transition in baseosmgr.

Reply:
If you pass a blank key `""`, then the worker always will submit it, and the only way to retrieve the result afterwards is via the `Process()`. On the other hand, if you have a key, you can retrieve it via `Process()` _or_ from the key.

Reply:
That was how I tried to capture both use cases.

Reply:
Would make sense to add some comments documenting that.

Reply:
Let me know if the docs make this clearer

Reply:
Comments you added look good